src/content/docs/kubernetes-pixie/kubernetes-integration/get-started/kubernetes-components.mdx

```diff
@@ -0,0 +1,129 @@
+  If you need to add an attribute to these samples, use `customAttributes`.
+</Callout>
+
+## Kubelet component [#nrk8s-kubelet]
+
+The Kubelet is the “Kubernetes agent”, a service that runs on every Kubernetes node. It's responsible for creating the containers as instructed by the control plane. Since it's the Kubelet who partners closely with the Container Runtime, it's the main source of infrastructure metrics for our integration, such as use of CPU, memory, disk, network, etc.
+
+`nrk8s-kubelet` runs as a DaemonSet where each instance gathers metric from the Kubelet running in the same node as it is. This component connects to the Kubelet using the Node IP. If this process fails, `nrk8s-kubelet` will fall back to reach the node through the API Server proxy. 
+
+<Callout variant="important">
+  If you have very large clusters, proxying many kubelets might increase the load in the API server.
+</Callout>
+
+To know if the API Server is being used as a proxy, check the logs and look for a similar message to this one:
+
+```
+Trying to connect to kubelet through API proxy
+```
+
+## Control plane component [#nrk8s-controlplane]
+
+New Relic deploys `nrk8s-controlplane` as a DaemonSet with `hostNetwork: true`. The configuration is structured to support autodiscover and static endpoints.
+A wide range of known defaults are provided as configuration entries, which allows you to tweak autodiscovery to your needs.
+
+You can have multiple endpoints per selector and add a probe mechanism, which automatically detects the correct one. You can use the same selector to try different configurations such as ports or protocols.
+
+Scraping configuration for the `etcd CP` component looks like the following where the same structure and features applies for all components:
+
+```yaml
+config:
+  etcd:
+    enabled: true
+    autodiscover:
+      - selector: "tier=control-plane,component=etcd"
+        namespace: kube-system
+        matchNode: true
+        endpoints:
+          - url: https://localhost:4001
+            insecureSkipVerify: true
+            auth:
+              type: bearer
+          - url: http://localhost:2381
+    staticEndpoint:
+      url: https://url:port
+      insecureSkipVerify: true
+      auth: {}
+```
+
+If `staticEndpoint` is set, the component will try to scrape it. If it can't hit the endpoint, the integration will fail so there are no silent errors when manual endpoints are configured.
+
+If `staticEndpoint` is not set, the component will iterate over the autodiscover entries looking for the first pod that matches the `selector` in the specified `namespace`, and optionally is running in the same node of the DaemonSet (if `matchNode` is set to `true`). When a pod is discovered, the component probes, issuing an http `HEAD` request, the endpoints are listed in order and scrapes the first successful probed one using the authorization type selected.
+
+The scraping logic is the same for the `etcd` component and other components.
```

Comment:
This is in-depth information intended for troubleshooting or fine-tuning an already advanced use case (control plane monitoring). I would move it to a specific page.

Reply:
I've removed it from here because @csongnr agrees with your comment.